zkevm-circuits/src/tx_circuit.rs

```diff
@@ -294,6 +296,17 @@ impl<F: Field> SubCircuitConfig<F> for TxCircuitConfig<F> {
         );
 
         // tx_id transition
+        meta.create_gate("tx_id starts with 1", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            cb.require_equal(
+                "tx_id == 1",
+                meta.query_advice(tx_table.tx_id, Rotation::cur()),
+                1.expr(),
+            );
+
+            cb.gate(meta.query_fixed(q_second, Rotation::cur()))
```

Comment:
We also need to check that this is not a padding row. In the edge case where no. of txs is zero, the `tx_id` at this row would be `0` (and this gate would fail).

Reply:
Even if the "real" user txs are none, we will pad `max_txs` padding tx in the end.

Reply:
@roynalnaruto Does the above comment answer your concern?

Reply:
the padding txs will have non-zero tx id, right? (as with the answer from question 19)

Reply:
yes. If the last real tx's id is `k`, then the padding txs' id will be [`k+1`, ... `max_txs`].